### PR TITLE
BOM-1543: Added python 3.8 in test matrix

### PR DESCRIPTION
**Pull request by iamsobanjaved**
- Removed testing of Django 1.11 ... 2.1 and added 3.0 instead
- Added travis worker of python 3.8
https://openedx.atlassian.net/browse/BOM-1543

### CODE REVIEW

File: search/result_processor.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def has_match(string):
     def decorate_matches(match_in, match_word):
         """ decorate the matches within the excerpt """
         matches = re.finditer(match_word, match_in, re.IGNORECASE)
-        for matched_string in set([match.group() for match in matches]):
+        for matched_string in {match.group() for match in matches}:
```
Comment (awais786) [inline search/result_processor.py:84]: purpose of this change ?

File: setup.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def is_requirement(line):
 
 setup(
     name='edx-search',
-    version='1.3.4',
+    version='1.3.5',
```
Comment (awais786) [inline setup.py:35]: You don't need to release it.

**Comment by mraarif** [comment]
as you can see from coverage report [here](https://codecov.io/gh/edx/edx-search/compare/b434d8968b016700c6e4b3d2e04dba2b2a471001...388a63b191d138ece6808d59365962869d408234/diff) there is still deprecated python 2 code left, maybe you want to spend more time you can refactor it.

File: .travis.yml [hunk 3]
```diff
@@ -2,26 +2,15 @@ language: python
...
-  include:
-    - python: 2.7
-      env: TOXENV=quality
-    - python: 2.7
-      env: TOXENV=django111
-    - python: 3.6
-      env: TOXENV=django22
-    - python: 3.7
-      env: TOXENV=django22
+  allow_failures:
```
Comment (morenol) [inline .travis.yml:12]: Is there a particular reason to allow failures of python 3.8? it seems to me that it is not needed anymore

Comment (iamsobanjaved) [reply .travis.yml]: Removed

**Comment by iamsobanjaved** [comment]
@jmbowman can i merge this PR as the [build](https://travis-ci.org/github/edx/edx-search/jobs/689755927) is passing but codecov is not being started with this error
`Error: 400 Client Error: Bad Request for url: https://storage.googleapis.com/codecov/v4/raw/2020-05-21/94071FF4B770728ED3A6D0D37C547B1D/948b557fc310ef564399f475fa43177099a033be/012f7dd3-1f79-47ba-b9ab-db7624a6ab46.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=GOOG1EQX6OZVJGHKK3633AAFGLBUCOOATRACRQRQF6HMSMLYUP6EAD6XSWAAY%2F20200521%2FUS%2Fs3%2Faws4_request&X-Amz-Date=20200521T184234Z&X-Amz-Expires=10&X-Amz-SignedHeaders=host&X-Amz-Signature=537f1f3c67d03149e853cfd3c7ef032751c83534868e94c9e71eb71fef01d771
`

**Comment by jmbowman** [comment]
A new version of codecov was just released to fix this: https://github.com/codecov/codecov-python/issues/251 .  Go ahead and upgrade it to 2.1.3 and see if it resolves the problem.

**Comment by awais786** [comment]
@iamsobanjaved its green now. Devops ticket and minor change in coverage file does the trick.

**Pull request merged**